packages/aws-cdk-lib/aws-route53/lib/record-set.ts

```
@@ -366,28 +399,42 @@ export class RecordSet extends Resource implements IRecordSet {
       throw new ValidationError(`setIdentifier must be between 1 and 128 characters long, got: ${props.setIdentifier.length}`, this);
     }
     if (props.setIdentifier && props.weight === undefined && !props.geoLocation && !props.region && !props.multiValueAnswer
-      && !props.cidrRoutingConfig) {
+      && !props.cidrRoutingConfig && !props.failover) {
       throw new ValidationError('setIdentifier can only be specified for non-simple routing policies', this);
     }
     if (props.multiValueAnswer && props.target.aliasTarget) {
       throw new ValidationError('multiValueAnswer cannot be specified for alias record', this);
     }
+    if (props.failover && props.multiValueAnswer) {
+      throw new ValidationError('Cannot use both failover and multiValueAnswer routing policies', this);
+    }
+    if (props.failover === Failover.PRIMARY && !props.healthCheck) {
+      Annotations.of(this).addWarningV2('@aws-cdk/aws-route53:primaryFailoverHealthCheck', 'PRIMARY failover record sets should include a health check for proper failover behavior');
```

This needs to fail hard, as the documentation is stating is a `must`. Feel free to add this warning but for Secondary types that have no health check (as that wont fail deployment, but might be important info to share to the implementers)